lib/SILOptimizer/IPO/CrossModuleOptimization.cpp

```diff
@@ -164,6 +166,52 @@ class InstructionVisitor : public SILCloner<InstructionVisitor> {
   }
 };
 
+static void emitRemark(StringRef remark, StringRef other, SILInstruction *i) {
+    if (!i->getModule().getASTContext().LangOpts.RemarkWhenFailedToSerialize)
+        return;
+
+    auto &ctx = i->getModule().getASTContext();
+    auto joined = remark.str() + other.str();
```

Comment:
Could you use `llvm::Twine` for string concatenation.

Reply:
I'm going to need to turn it into a string anyway to then turn it into a StringRef

Reply:
And then I think we need to globally allocate it in the AST context so that it's not deallocated by the time the diagnostic fires